### DISCUSSION TITLE
Noob questions and getting started

### DISCUSSION THREAD

**Discussion by fresh4** [coqui-ai/TTS#2425] [General Q&A]
I'm really new to AI/ML and have spent my spare time in the last week exploring this repository. The speech synthesis is really cool, but of course I want to synthesize a custom voice. Thankfully there is plenty of documentation, however I've found it a little bit confusing, so I'd like some clarification.

To start, I understand that I have to first create a tts model via my own dataset and associated metadata transcripts. I then have to train my vocoder on that tts model? I have some confusion between the two; I assume using a default/different vocoder on my own tts model won't give great results.

I admittedly don't have tons of data to train on; about 240 files spanning 15 minutes of audio. I can scrape up more but I was hoping I could start with this?

I've used the notebooks to tweak my config settings for audio, and use those settings when trying to run the training.

I've started by created the stats.npy file on my dataset, and then I try to run train_tts.py with my config (using the tacotron2 model and the ljspeech formatter), however it always crashes running out of memory. It is running with my GPU, a 4080 with 16gb vram, so it should be capable, but unless my batch size is reduced to around 2 or so, it won't run. Are there other settings or configuration I'm missing? Am I running the wrong training script?

Thanks in advance. I'm trying to wrap my head around this and wanna make sure I'm going about it 'correctly'.

**Comment by thorstenMueller** [comment]
Hi, tuning training params for a specific gpu power is a really specific topic, but maybe these two videos from my Youtube channel help on creating a custom TTS voice in general.
- https://youtu.be/4YT8WZT_x48
- https://youtu.be/bJjzSo_fOS8

**Reply by fresh4** [reply]
Hi! Funnily I had already seen your windows video. Though the longer video was pretty helpful, a great overview for anyone starting out! I see there's not many steps both here and in documentation in general regarding training a vocoder, so I'm wondering does a well trained tts model synthesize voice well enough? Or does it make its own vocoder depending on the model like you mentioned in one of your vids? I'm kinda hung up on the specifics.

I'm still sort of stuck on the whole batch size thing. I had done a test run with a smaller dataset (only 100 files) and would get the same memory errors until I disabled gradual training and just set the batch size to 32. However with this dataset, for some reason or another, it gives the same error with the same batch size. I can lower it to 2 or 4 (I think it breaks at 8) but I also have read that might just make it difficult for the training to be accurate anyway.
